### PR TITLE
adjustments on the gd32vf103

### PR DESCRIPTION
**Pull request by raffi-g**
This PR comprises some adjustments for the gd32vf103.

My goal here was to test the changes in my other PR #66759 using the longan_nano board and the gen_isr_table test.

I noticed that the alignment of the vector table is too small for the number of interrupts.

Also a lot of the soc-specific startup code is redundant. I therefore removed these parts and assured, that the soc-specific and the common code run both one after the other. Before, both parts were linked into the FW, but only one of them actually ran.

### CODE REVIEW

File: soc/riscv/riscv-privileged/gd32vf103/entry.S [hunk 1]
```diff
@@ -9,52 +9,27 @@
...
 
-	/* Jump to __reset */
-	tail __reset
-
-1:
-	j 1b
-
-.align 6
-trap_entry:
-	tail _isr_wrapper
```
Comment (soburi) [inline soc/riscv/riscv-privileged/gd32vf103/entry.S:60]: From the Nuclei ISA spec.
The mtvec register only can take an address that is aligned to 6.

https://doc.nucleisys.com/nuclei_spec/isa/core_csr.html#mtvec

Is it OK about this point?

Comment (raffi-g) [reply soc/riscv/riscv-privileged/gd32vf103/entry.S:60]: sorry for my late reply. I was on holiday break.

yes, this still works. In arch/riscv/core/isr.S you can see, that `_isr_wrapper` is aligned to `CONFIG_RISCV_TRAP_HANDLER_ALIGNMENT`. I've set this to 64, i.e. 2^6, in the last commit.

**Comment by soburi** [comment]
@raffi-g 
When I tried it on my longan nano, it seems that the UART communication is strange.
Could you check it?

**Comment by raffi-g** [comment]
> @raffi-g When I tried it on my longan nano, it seems that the UART communication is strange. Could you check it?

could you clarify a bit more what you mean by strange?

I've tested it again and can communicate with the board without a problem. For example the hello world example:

```
west build -b longan_nano samples/hello_world
```

listening in hterm with 115200 I get:

```
*** Booting Zephyr OS build 7c18f1b2f7aa ***
Hello World! longan_nano
```

**Comment by soburi** [comment]
@raffi-g 

> could you clarify a bit more what you mean by strange?

I checked again, and it works fine.
Sorry for confusing you.

**Review by soburi — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**